File: test/test-junit5/src/test/java/ru/tinkoff/kora/test/extension/junit5/mockito/MockWrappedParentTests.java
package ru.tinkoff.kora.test.extension.junit5.mockito;

import org.junit.jupiter.api.Test;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.internal.util.MockUtil;
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest;
import ru.tinkoff.kora.test.extension.junit5.TestComponent;
import ru.tinkoff.kora.test.extension.junit5.testdata.TestApplication;

import static org.junit.jupiter.api.Assertions.*;

@KoraAppTest(TestApplication.class)
public class MockWrappedParentTests {

    @TestComponent
    private TestApplication.SomeContainer someContainer;

    @Mock
    @TestComponent
    private TestApplication.SomeWrapped someWrappedInterface;
    @Mock
    @TestComponent
    private TestApplication.SomeChild someChildInterface;
    @Mock
    @TestComponent
    private TestApplication.CustomWrapper customWrapper;

    @Test
    void wrappedMocked() {
        MockUtil.isMock(someChildInterface);
        assertNotNull(customWrapper);

        Mockito.when(someWrappedInterface.toString()).thenReturn("12345");

        assertSame(someWrappedInterface, someContainer.wrapped());

        assertEquals("12345", someWrappedInterface.toString());
        assertEquals("12345", someContainer.wrapped().toString());
    }
}

File: test/test-junit5/src/test/kotlin/ru/tinkoff/kora/test/extension/junit5/kotlin/mockk/MockWrappedParentTests.kt
package ru.tinkoff.kora.test.extension.junit5.kotlin.mockk

import io.mockk.every
import io.mockk.impl.annotations.MockK
import io.mockk.isMockKMock
import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test
import ru.tinkoff.kora.test.extension.junit5.KoraAppTest
import ru.tinkoff.kora.test.extension.junit5.TestComponent
import ru.tinkoff.kora.test.extension.junit5.testdata.TestApplication

@KoraAppTest(TestApplication::class)
class MockWrappedParentTests(
    @MockK @TestComponent val someWrappedInterface: TestApplication.SomeWrapped,
    @TestComponent val someContainer: TestApplication.SomeContainer,
    @MockK @TestComponent val someChildInterface: TestApplication.SomeChild,
    @MockK @TestComponent val customWrapper: TestApplication.CustomWrapper
) {

    @Test
    fun wrappedMocked() {
        assertTrue(isMockKMock(someChildInterface))
        every { someWrappedInterface.toString() } returns ("12345")
        assertNotNull(customWrapper)

        assertSame(someWrappedInterface, someContainer.wrapped())

        assertEquals("12345", someWrappedInterface.toString())
        assertEquals("12345", someContainer.wrapped().toString())
    }
}
